### ISSUE TITLE
[Apple M1] Compile and pip errors: Generating _shaders.c ninja: build stopped: subcommand failed.

### ISSUE THREAD

**Issue by kevinbein** [datoviz/datoviz#47]
This projects looks great so I wanted to try and compile it on my MacBook Pro M1 (Silicon). I get that this project is still in early alpha and that there are no arm releases but I will leave this feedback here anyway since others may stumble upon the same problems:

- Installing via pip (even though it is targeted at x86_64) yields the following error: `ERROR: datoviz-0.1.0a1-cp38-cp38-macosx_10_14_x86_64.whl is not a supported wheel on this platform.`. I am not sure if this is related to the build or if the .wheel format is actually not supported on M1s.
- Compiling the source via `./manage.sh build` throws a bunch of errors:

```
...
FAILED: _deps/cglm-build/CMakeFiles/cglm.dir/src/mat4.c.o
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -DCGLM_EXPORTS -Dcglm_EXPORTS -I/opt/datoviz/build/_deps/cglm-src/include -I/opt/datoviz/build/_deps/cglm-src/src -g -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.0.sdk -fPIC -Wall -Werror -O3 -std=gnu11 -MD -MT _deps/cglm-build/CMakeFiles/cglm.dir/src/mat4.c.o -MF _deps/cglm-build/CMakeFiles/cglm.dir/src/mat4.c.o.d -o _deps/cglm-build/CMakeFiles/cglm.dir/src/mat4.c.o -c /opt/datoviz/build/_deps/cglm-src/src/mat4.c
In file included from /opt/datoviz/build/_deps/cglm-src/src/mat4.c:8:
In file included from /opt/datoviz/build/_deps/cglm-src/src/../include/cglm/cglm.h:18:
In file included from /opt/datoviz/build/_deps/cglm-src/src/../include/cglm/affine.h:41:
In file included from /opt/datoviz/build/_deps/cglm-src/src/../include/cglm/affine-mat.h:30:
/opt/datoviz/build/_deps/cglm-src/src/../include/cglm/simd/neon/affine.h:89:29: error: variable 'r3' set but not used [-Werror,-Wunused-but-set-variable]
  glmm_128      r0, r1, r2, r3, x0;
                            ^
1 error generated.
[50/140] Generating _shaders.c
ninja: build stopped: subcommand failed.
```

Is development on this project still ongoing? Thanks for the effort anyway!

**Comment by rossant** [comment]
Thanks for your interest! Yes, the project is still in active development, although most of the development occurs in a [separate repository for now](https://github.com/datoviz/datoviz-distributed).

I do have a MacBook Pro M1 so I'll investigate the issue and report back.

**Comment by rossant** [comment]
I could reproduce the issue on my Mac. I pushed a quick fix (just updating the versions of cglm and glfw dependencies), it did the trick for me, hopefully it will for you too!

**Comment by kevinbein** [comment]
Thank you for the quick update and sorry for my late response. I compiled it and that part worked. However, I do get a different error when I run `./manage.sh demo scatter`:

```
☁  datoviz [main] ⚡  ./manage.sh demo scatter
16:38:03.107 I             main.c:0051: running scatter plot demo with 50000 points
16:38:03.150 E     vklite_utils.h:0280: validation layer: vkCreateInstance: Found drivers that contain devices which support the portability subset, but the portability enumeration bit was not set! Applications that wish to enumerate portability drivers must set the VK_INSTANCE_CREATE_ENUMERATE_PORTABILITY_BIT_KHR bit in the VkInstanceCreateInfo flags and enable the VK_KHR_portability_enumeration instance extension.
16:38:03.150 E     vklite_utils.h:0280: validation layer: vkCreateInstance: Found no drivers!
16:38:03.150 E     vklite_utils.h:0068: VkResult is ERROR_INCOMPATIBLE_DRIVER in /opt/datoviz/src/vklite_utils.h at line 68
16:38:03.150 E     vklite_utils.h:0068: VkResult is ERROR_EXTENSION_NOT_PRESENT in /opt/datoviz/src/vklite_utils.h at line 68
```

As I am learning Vulkan myself right now, I already stumbled upon this problem. The solution therefor is:

- Setting  `VkInstanceCreateInfo.flags` to `VK_INSTANCE_CREATE_ENUMERATE_PORTABILITY_BIT_KHR`
- Adding to `VkInstanceCreateInfo.ppEnabledExtensionNames` the value of `VK_KHR_PORTABILITY_ENUMERATION_EXTENSION_NAME`

([https://github.com/KhronosGroup/MoltenVK#developing-vulkan-applications-for-macos-ios-and-tvos]https://github.com/KhronosGroup/MoltenVK#developing-vulkan-applications-for-macos-ios-and-tvos)

**Comment by rossant** [comment]
Thanks. I just pushed a new fix, does it work?

**Comment by kevinbein** [comment]
Yes, this worked perfectly!

Although for it to work with my setup I had to additionally do the following:

- I am using brew to manage my python versions. MacOS (I am on Ventura) has, by default, python in version 3.10 installed. I am using python 3.11 so I replaced `python3` in `manage.sh` with `python3.11` (at least that is how I linked it). By now I don't think this has a direct impact but I wanted to make sure the versions match since ...
- ... I also had to install the dependency `cython` with `python3.11 -m pip install python`
- Since my Vulkan installation is located in a special folder, I also had to do an `export VULKAN_SDK="/opt/VulkanSDK/1.3.236.0/macOS"` (and adding it to my `~/.zshrc` file) prior to running `./manage.sh python`

With this, the scatter demo and python example both work fantastic.

Thank you for your help!
